File: Decorator/Golang/ColorfulDecorator.go
package main

import "fmt"

type ColorfulDecorator struct {
	PrinterDecorator
}

func (colorfulDecorator ColorfulDecorator) print()  {
	colorfulDecorator.PrinterDecorator.print()
	fmt.Println("Printed with colorful color")
}

func NewColorfulDecorator(printer IPrinter) ColorfulDecorator {
	return ColorfulDecorator{PrinterDecorator{printer}}
}

File: Decorator/Dart/ColorfulDecorator.dart
import 'IPrinter.dart';
import 'PrinterDecorator.dart';

class ColorfulDecorator extends PrinterDecorator {
  ColorfulDecorator(IPrinter printer) : super(printer);

  @override
  void printing() {
    super.printing();
    print("Printed with colorful color");
  }
}
